doc/Lorre's Block Fetching Algorithm

```diff
@@ -0,0 +1,96 @@
+Lorre's Block Fetching Algorithm:
+
+The following logic is implemented via the processBlocks function, which is a private helper of the
+more general getBlocks function.
+
+Initially, when Lorre is first run with an empty database, it connects to the head block on the chain
+at level n, and will add blocks at levels 0..n to the chain. If there is no fork at all,
+the next iteration of Lorre will run. If there are no new blocks, nothing is added to the database.
+If there are new blocks, let the head block's level be n_2. Then, blocks n+1, n+2...n_2 will be
+added to the database. Inductively, if there are never any forks, then at iteration i of Lorre,
+Lorre will add blocks n_{i-1} + 1, n_{i-1} + 2...n_i to the database.
+
+ASSUMPTION: In a non-forked chain, there can only be one block at each level. [i]
+
+------------------------------------------------------------------------------------------------------
+
+Lorre's algorithm for fetching blocks that occur as a result of a fork:
+
+The following logic is implemented via the followFork function, which is a private helper of the
+more general getBlocks function.
+
+In the normal case, we place blocks into a blocks table, with the fields being all of the associated
+metadata. In the forked case, we will have an invalidated blocks table to invalidate blocks that are
```

Comment:
Not clear to me if, in case of a detected fork, all blocks at any forked level should be present in the `Invalidated` table... or only the invalidated ones?

Reply:
only blocks that have been invalidated at least once will be in the invalidated_blocks table. The idea is at level n, a block is the "true" block if either it is not in the invalidated blocks table, or it is in the invalidated blocks table with it's is_invalidated field set to false. The assumed invariant is that, at any given point in time, there is at most one block at a given level that has is_invalidated set to false in the invalidated blocks table